File: Ola/Array/Day 4/search-insert.js
//Solution
var searchInsert = function(nums, target) {
  let left= 0;
  let right= nums.length;
  while (left <=right){
      const middle = Math.floor((left + right)/2)
      if (nums[middle]<target){
          left+=1
      }
      else{
        right -=1  
      }
  }
  return left
};
console.log(`The answer is ${searchInsert([1,3,5,6], 5)}`)

File: Tope/Day4/searchInsert.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        lens = len(nums)
        if(lens == 0): return 0

        for i,n in enumerate(nums):
            if(n >= target):
                return i

        return lens

input_list = [3,12,14,17]
ob1 = Solution()
print(ob1.searchInsert(input_list, 15))